crypto/x509/x_pubkey.c

```diff
@@ -85,12 +86,15 @@ static int x509_pubkey_ex_populate(ASN1_VALUE **pval, const ASN1_ITEM *it)
             || (pubkey->public_key = ASN1_BIT_STRING_new()) != NULL);
 }
 
-static int x509_pubkey_ex_new(ASN1_VALUE **pval, const ASN1_ITEM *it)
+
+static int x509_pubkey_ex_new_ex(ASN1_VALUE **pval, const ASN1_ITEM *it,
```

Comment:
Are internal functions with _ex_ twice in the names helpful?
I'm dubious about functions where the _ex_ is in the middle of the name.
They don't seem to add anything and make reading the name more difficult.

Reply:
The first `ex` isn't for `extended`, but for `external`, i.e. reflects "EXTERN" in the type `ASN1_EXTERN_FUNCS`
The pattern was taken from here, which has existed since half forever:

https://github.com/openssl/openssl/blob/74613e8c97be6c310152f7cc6d8162f3ccc723a4/crypto/x509/x_name.c#L74-L82

Reply:
I hesitated over this too. But as @levitte points out the first "ex" is for external and so has a different meaning. Our policy is to always add an "ex" for function extensions so I went with the name that conforms to policy.